native/src/main/java/io/ballerina/stdlib/io/nativeimpl/RecordChannelUtils.java

```diff
@@ -142,43 +142,57 @@ public static Object getAllRecords(BObject channel, int skipHeaders, BTypedesc t
         }
         try {
             if (describingType.getTag() == TypeTags.RECORD_TYPE_TAG) {
+                if (skipHeaders == 1) {
+                    return IOUtils.createError("Parameter `skipHeaders` cannot be used with record data mapping. ");
+                }
```

Comment:
Do we need to return the error here, as we don't care abt skipHeader for the return record

Reply:
Yes. We can remove this. Shall we print a warning log and continue?

Reply:
updated in [a336bb2](https://github.com/ballerina-platform/module-ballerina-io/pull/876/commits/a336bb2c7df06843183fab4317574f7af271495a)